aspnetcore/blazor/performance/profiling.md

```diff
@@ -0,0 +1,226 @@
+description: Learn about performance profiling and diagnostic counters in ASP.NET Core Blazor WebAssembly apps.
+monikerRange: '>= aspnetcore-10.0'
+ms.author: riande
+ms.custom: mvc
+ms.date: 04/16/2025
+uid: blazor/performance/profiling
+---
+# ASP.NET Core Blazor WebAssembly performance profiling and diagnostic counters
+
+<!-- UPDATE 10.0 - Activate ...
+
+[!INCLUDE[](~/includes/not-latest-version.md)]
+
+-->
+
+This article describes performance profiling tools and diagnostic counters for Blazor WebAssembly apps.
+
+## Prerequisite
+
+Install the [.NET WebAssembly build tools](xref:blazor/tooling/webassembly#net-webassembly-build-tools):
+
+```dotnetcli
+dotnet workload install wasm-tools
+```
+
+## Browser developer tools
+
+App code can be manually profiled using the performance profiler in a browser's developer tools console.
+
+Built-in performance counters are available to track:
+
+* [Ahead-of-time (AOT) compilation](xref:blazor/tooling/webassembly#ahead-of-time-aot-compilation)
+* Code interpolation
+* [JIT (Just-In-Time) interpolation](https://developer.mozilla.org/docs/Glossary/Just_In_Time_Compilation) 
+* Call specification (":::no-loc text="callspec":::", sequence and timing of function calls) and instrumentation
+
+Enable integration with the browser's developer tools profiler using the `<WasmProfilers>` property in the app's project file (`.csproj`). Include the additional properties in the following table.
+
+Property | Default | Set value to&hellip; | Description
+--- | :---: | :---: | ---
+`<WasmProfilers>` | No value | `browser` | Mono profilers to use. Potential values are "`browser`" and "`log`". To use both, separate the values with a semicolon. The `browser` profiler enables integration with the browser's developer tools profiler.
+`<RunAOTCompilation>`| `false` | `true` | Controls AOT compilation.
+`<RunAOTCompilationAfterBuild>` | `false` | `true` | Controls AOT compilation after build. By default, it's run only for publish.
+`<WasmNativeStrip>` | `true` | `false` | Controls stripping the native executable.
+`<WasmNativeDebugSymbols>` | `true` | `true` | Controls building with native debug symbols.
+`<WasmBuildNative>` | `false` | `true` | Controls building the native executable.
+
+```xml
+<PropertyGroup>
+  <WasmProfilers>browser;</WasmProfilers>
+  <RunAOTCompilation>true</RunAOTCompilation>
+  <RunAOTCompilationAfterBuild>true</RunAOTCompilationAfterBuild>
+  <WasmNativeStrip>false</WasmNativeStrip>
+  <WasmNativeDebugSymbols>true</WasmNativeDebugSymbols>
+  <WasmBuildNative>true</WasmBuildNative>
+</PropertyGroup>
+```
+
+Add Blazor start configuration in `wwwroot/index.html`, using the [fingerprinted location of the Blazor WebAssembly script](xref:blazor/fundamentals/static-files#fingerprint-client-side-static-assets-in-standalone-blazor-webassembly-apps). In the following example, the `sampleIntervalMs` option is set to 10 seconds, which is the default setting if `sampleIntervalMs` isn't specified:
```

Comment:
You don't have to do `builder.withConfig` anymore, the MSbuild properties would suffice.

Reply:
Are you asking to remove the following `Blazor.start` configuration? If so, how is the sample interval set?

Reply:
Nevermind ... I think you just mean that this text should change to make this optional if they want to set the interval. I'll place this into a new section.

Reply:
I mean that you could remove it and it should flow from MSbuild properties

Reply:
> interval should flow from MSbuild properties

What's the property name/configuration? Is it .........

```xml
<SampleIntervalMs>10</SampleIntervalMs>
```

Reply:
... and will the same thing be true for `callSpec`? Instead of `Blazor.start`, will the following work? ...

```xml
<CallSpec>N:BlazorSample</CallSpec>
```

Reply:
so MSBuild for sample interval will be ....

```xml
<WasmProfilers>browser:sampleIntervalMs=10;</WasmProfilers>
```

Reply:
Latest commit sets up guidance for both scenarios and both options. 🎉